presto-spi/src/main/java/com/facebook/presto/spi/predicate/TupleDomain.java

```diff
@@ -263,31 +263,29 @@ public boolean isNone()
                 commonColumns.retainAll(domain.getDomains().get().keySet());
             }
         }
+        return columnWiseUnion(commonColumns, tupleDomains);
+    }
 
-        // group domains by column (only for common columns)
-        Map<T, List<Domain>> domainsByColumn = new HashMap<>(tupleDomains.size());
+    /**
+     * Returns a TupleDomain in which specified column Domains are unioned together.
+     */
+    public static <T> TupleDomain<T> columnWiseUnion(Set<T> columns, Iterable<TupleDomain<T>> tupleDomains)
```

Comment:
As a generic piece, when called directly, its missing a bunch of the handling for edge conditions (see the checks that are applied in the original columnWiseUnion). If those conditions are not also checked, then this produces wrong results.